File: src/main/java/com/severett/chargerapp/data/InMemoryChargingSessionRepo.java
package com.severett.chargerapp.data;

import com.severett.chargerapp.model.ChargingSession;
import org.springframework.stereotype.Repository;

import java.util.Map;
import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;

@Repository
public class InMemoryChargingSessionRepo implements ChargingSessionRepo {

    private final Map<UUID, ChargingSession> sessionMap;

    public InMemoryChargingSessionRepo() {
        sessionMap = new ConcurrentHashMap<>();
    }

    @Override
    public ChargingSession save(ChargingSession chargingSession) {
        return sessionMap.put(chargingSession.getId(), chargingSession);
    }

    @Override
    public Optional<ChargingSession> findById(UUID id) {
        return Optional.ofNullable(sessionMap.get(id));
    }

    @Override
    public Iterable<ChargingSession> findAll() {
        return sessionMap.values();
    }

}

File: src/main/kotlin/com/severett/chargerapp/data/InMemoryChargingSessionRepo.kt
package com.severett.chargerapp.data

import com.severett.chargerapp.model.ChargingSession
import org.springframework.stereotype.Repository
import java.util.*
import java.util.concurrent.ConcurrentHashMap

@Repository
class InMemoryChargingSessionRepo : ChargingSessionRepo {

    private val sessionMap = ConcurrentHashMap<UUID, ChargingSession>()

    override fun save(chargingSession: ChargingSession): ChargingSession {
        sessionMap[chargingSession.id] = chargingSession
        return chargingSession
    }

    override fun findById(id: UUID): Optional<ChargingSession> = Optional.ofNullable(sessionMap[id])

    override fun findAll(): Iterable<ChargingSession> = sessionMap.values

}
